src/activity_handlers.cpp

```diff
@@ -851,11 +852,18 @@ void activity_handlers::start_fire_lens_do_turn( player_activity *act, player *p
         float previous_natural_light_level = act->values.back();
         act->values.pop_back();
         act->values.push_back(natural_light_level);
-        iuse tmp;
+        item &lens_item = p->i_at(act->position);
+        const auto *usef = lens_item.type->get_use( "extended_firestarter" );
+        if( usef == nullptr || usef->get_actor_ptr() == nullptr ) {
+            add_msg( m_bad, "You have lost the item you were using as a lens." );
+            p->cancel_activity();
```

Comment:
Return would be required here, otherwise the following use of `usef` will crash.